gui/default/syncthing/core/syncthingController.js

```diff
@@ -1732,21 +1732,21 @@ angular.module('syncthing.core')
         };
 
         $scope.setAllDevicesPause = function (pause) {
-            for (var id in $scope.devices) {
-                $scope.devices[id].paused = pause;
-            };
-            $scope.config.devices = deviceList($scope.devices);
+            $scope.otherDevices().forEach(function (device) {
+                device.paused = pause;
+            });
+            $scope.config.devices = deviceList($scope.otherDevices());
```

Comment:
In the `foreach` you are modifying in place. That might work, i.e. modifies the corresponding object in `$scope.devices` directly. I'd still do that explicitly, i.e. modify `$scope.devices[device.DeviceID] = paused`. Then in the end you still need assign all devices to the config not just the "others".

Reply:
I have fixed the all devices assigning part, but if I try to do the explicit modification, e.g.

```
$scope.devices[device.DeviceID].paused = pause;
```

or

```
$scope.devices[device].paused = pause;
```

then I get

```
Cannot set property 'paused' of undefined
```

errors.

Reply:
Because I messed up the camelCase that is used in JSland: `$scope.devices[device.deviceID].paused = pause;`. Generally the console that shows you the error also lets you interact with the objects, i.e. you can see what `device` is and what its members are named.

Reply:
And I actually searched for a similar syntax in the rest of the code, and found it, but failed to notice the case difference too :hushed:. I just use the DevTools in Chromium, but my JS console skills are extremely basic.

I think that I have fixed everything now.